### PR TITLE
feat(bulk-import): allow skipping the CODEOWNERS dry-run check [RHIDP-3339]

### PR DESCRIPTION
**Pull request by rm3l**
**What does this PR do / why we need it:**

This PR is an addendum to https://github.com/janus-idp/backstage-plugins/pull/1979
Currently, the dry-run operation always checks for the presence of a CODEOWNERS file. We want to skip this check if the client does not request it explicitly.
This is needed in the UI. Otherwise, the UI will not allow creating a bulk import PR even if the user did not select the checkbox to use the CODEOWNERS file as entity owner.

To support this, a new `codeOwnersFileAsEntityOwner` field has been added to each item in the ImportRequest object.

**Which issue(s) this PR fixes:**

ref https://issues.redhat.com/browse/RHIDP-3339

**PR acceptance criteria:**

- [x] Unit tests

- [ ] Integration tests

- [x] Documentation 

**How to test changes / Special notes to the reviewer:**

/cc @debsmita1 @ciiay

### CODE REVIEW

**Comment by rm3l** [comment]
CI failing due to dirty `yarn.lock` file: https://github.com/janus-idp/backstage-plugins/actions/runs/10473193033/job/29004508469?pr=2068
Will rebase once https://github.com/janus-idp/backstage-plugins/pull/2069 is merged.

**Comment by debsmita1** [comment]
Tested this PR locally
It checks for the CODEOWNERS file only when the `Use CODEOWNERS file as Entity Owner` checkbox is checked

LGTM

will wait for @PatAKnight 's approval

**Review by PatAKnight — APPROVED** [review]
lgtm

`CODEOWNERS_FILE_NOT_FOUND_IN_REPO`
```
curl -X POST http://localhost:7007/api/bulk-import/imports?dryRun=true -d '[{"catalogEntityName": "bulk-import-test","codeOwnersFileAsEntityOwner": true,"repository": {"url": "https://github.com/test-org-pat/bulk-import-test", "defaultBranch": "main"}, "catalogInfoContent": "---\napiVersion: backstage.io/v1alpha1\nkind: Component\nmetadata:\n  name: bulk-import-test\n  annotations:\n    github.com/project-slug: test-org-pat/bulk-import-test\nspec:\n  type: other\n  lifecycle: unknown\n  owner: user:default/pataknight\n---\n"}]' -H "Content-Type: application/json" -H "Authorization: Bearer $token" -v | jq

[
  {
    "errors": [
      "CODEOWNERS_FILE_NOT_FOUND_IN_REPO"
    ],
    "catalogEntityName": "bulk-import-test",
    "repository": {
      "url": "https://github.com/test-org-pat/bulk-import-test",
      "name": "bulk-import-test",
      "organization": "test-org-pat"
    }
  }
]
```

No errors with same repo because `codeOwnersFileAsEntityOwner` is set to `false`
```
curl -X POST http://localhost:7007/api/bulk-import/imports?dryRun=true -d '[{"catalogEntityName": "bulk-import-test","codeOwnersFileAsEntityOwner": false,"repository": {"url": "https://github.com/test-org-pat/bulk-import-test", "defaultBranch": "main"}, "catalogInfoContent": "---\napiVersion: backstage.io/v1alpha1\nkind: Component\nmetadata:\n  name: bulk-import-test\n  annotations:\n    github.com/project-slug: test-org-pat/bulk-import-test\nspec:\n  type: other\n  lifecycle: unknown\n  owner: user:default/pataknight\n---\n"}]' -H "Content-Type: application/json" -H "Authorization: Bearer $token" -v | jq

[
  {
    "errors": [],
    "catalogEntityName": "bulk-import-test",
    "repository": {
      "url": "https://github.com/test-org-pat/bulk-import-test",
      "name": "bulk-import-test",
      "organization": "test-org-pat"
    }
  }
]
```

`REPO_EMPTY` and `CODEOWNERS_FILE_NOT_FOUND_IN_REPO`
```
curl -X POST http://localhost:7007/api/bulk-import/imports?dryRun=true -d '[{"catalogEntityName": "catalog-test-repo","codeOwnersFileAsEntityOwner": true,"repository": {"url": "https://github.com/test-org-pat/catalog-test-repo", "defaultBranch": "main"}, "catalogInfoContent": "---\napiVersion: backstage.io/v1alpha1\nkind: Component\nmetadata:\n  name: catalog-test-repo\n  annotations:\n    github.com/project-slug: test-org-pat/catalog-test-repo\nspec:\n  type: other\n  lifecycle: unknown\n  owner: user:default/pataknight\n---\n"}]' -H "Content-Type: application/json" -H "Authorization: Bearer $token" -v | jq

[
  {
    "errors": [
      "CODEOWNERS_FILE_NOT_FOUND_IN_REPO",
      "REPO_EMPTY"
    ],
    "catalogEntityName": "catalog-test-repo",
    "repository": {
      "url": "https://github.com/test-org-pat/catalog-test-repo",
      "name": "catalog-test-repo",
      "organization": "test-org-pat"
    }
  }
]
```

Just `REPO_EMPTY` because `codeOwnersFileAsEntityOwner` set to `false`
```
curl -X POST http://localhost:7007/api/bulk-import/imports?dryRun=true -d '[{"catalogEntityName": "catalog-test-repo","codeOwnersFileAsEntityOwner": false,"repository": {"url": "https://github.com/test-org-pat/catalog-test-repo", "defaultBranch": "main"}, "catalogInfoContent": "---\napiVersion: backstage.io/v1alpha1\nkind: Component\nmetadata:\n  name: catalog-test-repo\n  annotations:\n    github.com/project-slug: test-org-pat/catalog-test-repo\nspec:\n  type: other\n  lifecycle: unknown\n  owner: user:default/pataknight\n---\n"}]' -H "Content-Type: application/json" -H "Authorization: Bearer $token" -v | jq

[
  {
    "errors": [
      "REPO_EMPTY"
    ],
    "catalogEntityName": "catalog-test-repo",
    "repository": {
      "url": "https://github.com/test-org-pat/catalog-test-repo",
      "name": "catalog-test-repo",
      "organization": "test-org-pat"
    }
  }
]
```

**Pull request merged**
